code/datums/components/attachment_handler.dm

```diff
@@ -293,7 +293,9 @@
 		if(CHECK_BITFIELD(attachment_data[FLAGS_ATTACH_FEATURES], ATTACH_SAME_ICON))
 			icon_state = attachment.icon_state
 			icon = attachment.icon
-		overlay = image(icon, parent_item, icon_state, attachment_data[ATTACHMENT_LAYER])
+		overlay = image(icon, parent_item, icon_state)
+		if(CHECK_BITFIELD(attachment_data[FLAGS_ATTACH_FEATURES], ATTACH_SAME_ICON))
+			overlay.overlays += attachment.overlays
```

Comment:
Same check twice

Reply:
i cant merge both checks since overlay takes params from the first check, then is modified in the second check. If i merged them overlay would be generated with the wrong params, or overlays would be added to it then immediately the var would be overwritten